model.py

```diff
@@ -0,0 +1,118 @@
+import pandas as pd
+import matplotlib.pyplot as plt
+import numpy as np
+
+from sklearn.model_selection import train_test_split
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import mean_squared_error
+
+from preprocessing import preprocessing
+
+all_data = preprocessing()
+
+# print(all_data)
+'''
+    Location        DATE    Alamosa     Denver      Grand    Springs
+    0         2003-01-01  24.985507  33.523256  26.692130  31.296296
+    1         2003-01-02  15.527778  37.068966  27.435374  31.213178
+    2         2003-01-03  22.891156  47.344828  28.850340  43.343537
+    3         2003-01-04  26.864583  39.812500  31.074830  38.122449
+    4         2003-01-05  30.284722  41.914286  32.323529  39.490000
+    ...              ...        ...        ...        ...        ...
+    7300      2022-12-27  31.413793  48.848485  31.000000  49.303571
+    7301      2022-12-28  36.068966  39.130435  32.466292  41.100000
+    7302      2022-12-29  21.800000  25.578947  24.492754  28.648649
+    7303      2022-12-30  20.327586  26.424242  28.638889  27.965517
+    7304      2022-12-31  30.800000  32.794118  31.682540  44.338710
+    '''
+
+all_data['DATE'] = pd.to_datetime(all_data['DATE'])
+
+#Adding lag and rolling mean features (Feature engineering to help with the models accuracy)
+lag_featues = pd.DataFrame()
+rolling_mean_feature = pd.DataFrame()
+rolling_median_feature = pd.DataFrame()
+rolling_std_feature = pd.DataFrame()
+differential_feature = pd.DataFrame()
+
+# Shift() function in pandas creates these shifted or lag features from time series data.
+# Adding a lag of 1 day
+for lag in range(1,2):
+    for location in ['Alamosa','Grand','Springs','Denver']:
+        lag_featues[f"{location}_lag_{lag}"] = all_data[location].shift(lag)

```

Comment:
This is interesting, can you please give more detail explaining why you did this?

Reply:
Lag features help the model learn from patterns that happened in the past. In this case we are taking into account yesterdays temperature to predict tomorrow temperature. I experimented with using 1,3,7,14 and 30 days. 1 day provided me with the best results. A found a good example on this article.
"For example, it can be beneficial to generate features for sales that happened in previous days at 4:00 p.m. if you want to predict similar sales at 4:00 p.m. the next day." (https://medium.com/data-science-at-microsoft/introduction-to-feature-engineering-for-time-series-forecasting-620aa55fcab0)